starknet/src/utils/types.cairo

```diff
@@ -438,74 +215,8 @@ impl StorageAccessFelt252Array of StorageAccess<Array<felt252>> {
         }
     }
 
-    fn size_internal(value: Array<felt252>) -> u8 {
-        if value.len() == 0 {
-            return 1;
-        }
-        1_u8 + StorageAccess::<felt252>::size_internal(*value[0]) * value.len().try_into().unwrap()
-    }
-}
-
-impl StorageAccessStrategy of StorageAccess<Strategy> {
-    // #[inline(always)]
-    fn read(address_domain: u32, base: StorageBaseAddress) -> SyscallResult<Strategy> {
-        StorageAccessStrategy::read_at_offset_internal(address_domain, base, 0)
-    }
-    // #[inline(always)]
-    fn write(address_domain: u32, base: StorageBaseAddress, value: Strategy) -> SyscallResult<()> {
-        StorageAccessStrategy::write_at_offset_internal(address_domain, base, 0, value)
-    }
-
-    fn read_at_offset_internal(
-        address_domain: u32, base: StorageBaseAddress, offset: u8
-    ) -> SyscallResult<Strategy> {
-        Result::Ok(
-            Strategy {
-                address: StorageAccess::read_at_offset_internal(
-                    address_domain,
-                    storage_base_address_from_felt252(
-                        storage_address_from_base_and_offset(base, 0_u8).into()
-                    ),
-                    offset
-                )?,
-                params: StorageAccess::read_at_offset_internal(
-                    address_domain,
-                    storage_base_address_from_felt252(
-                        storage_address_from_base_and_offset(base, 1_u8).into()
-                    ),
-                    offset
-                )?
-            }
-        )
-    }
-
-    fn write_at_offset_internal(
-        address_domain: u32, base: StorageBaseAddress, offset: u8, value: Strategy
-    ) -> SyscallResult<()> {
-        // Write value.address at offset 0
-        StorageAccess::write_at_offset_internal(
-            address_domain,
-            storage_base_address_from_felt252(
-                storage_address_from_base_and_offset(base, 0_u8).into()
-            ),
-            offset,
-            value.address
-        );
-
-        // Write value.params at offset 1
-        StorageAccess::write_at_offset_internal(
-            address_domain,
-            storage_base_address_from_felt252(
-                storage_address_from_base_and_offset(base, 1_u8).into()
-            ),
-            offset,
-            value.params
-        )
-    }
-
-    fn size_internal(value: Strategy) -> u8 {
-        // Add 1 for the strategy address
-        StorageAccess::size_internal(value.params) + 1
+    fn size() -> u8 {
+        255_u8
```

Comment:
why 255?

Reply:
now size() doesnt take an argument so there is no way to know the size of dynamic types. I spoke to eni about this today and we agreed it made the most sense to put the max size possible. 

Ill add a comment to clarify

Reply:
ugh yeah this needs to be take a snapshot... gotcha